source/Octopus.Tentacle.Tests/Diagnostics/Metrics/KubernetesAgentMetricsFixture.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using FluentAssertions;
+using Newtonsoft.Json;
+using NSubstitute;
+using NSubstitute.ExceptionExtensions;
+using NUnit.Framework;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Kubernetes.Diagnostics;
+
+namespace Octopus.Tentacle.Tests.Diagnostics.Metrics
+{
+    public class KubernetesAgentMetricsFixture
+    {
+        readonly ISystemLog systemLog = Substitute.For<ISystemLog>();
+        
+        [Test]
+        public void CanAddMetricToAnEmptyPersistenceMap()
+        {
+            MockPersistenceProvider persistenceProvider = new();
+            var sut = new KubernetesAgentMetrics(persistenceProvider, systemLog);
+
+            var eventTimestamp = DateTimeOffset.Now;
+            sut.TrackEvent("Killed", "NFS Pod", eventTimestamp);
+
+            persistenceProvider.Content.Keys.Should().ContainSingle(entry => entry.Equals("Killed"));
+            var item = JsonConvert.DeserializeObject<EventJsonEntry>(persistenceProvider.Content["Killed"]);
+            item.Should().BeEquivalentTo(new EventJsonEntry("NFS Pod", new List<DateTimeOffset>() { eventTimestamp }));
+        }
+
+        // [Test]
```

Comment:
Can we potentially uncomment this and add the `[ignore]` attribute like the others?

Reply:
removed